src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionSetComparer.cs

```diff
@@ -1,77 +1,52 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Language.Intellisense;
 using Microsoft.VisualStudio.Text;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
 {
     internal class SuggestedActionSetComparer : IComparer<SuggestedActionSet>
     {
-        private readonly SnapshotPoint? _targetPoint;
+        private readonly TextSpan? _targetSpan;
 
-        public SuggestedActionSetComparer(SnapshotPoint? targetPoint)
+        public SuggestedActionSetComparer(TextSpan? targetSpan)
         {
-            _targetPoint = targetPoint;
+            _targetSpan = targetSpan;
         }
 
-        private static int Distance(Span? textSpan, SnapshotPoint? targetPoint)
+        private static int Distance(Span? textSpan, TextSpan? targetSpan)
         {
             // If we don't have a text span or target point we cannot calculate the distance between them
-            if (!textSpan.HasValue || !targetPoint.HasValue)
+            if (!textSpan.HasValue || !targetSpan.HasValue)
             {
                 return int.MaxValue;
             }
 
             var span = textSpan.Value;
-            var position = targetPoint.Value.Position;
+            var target = targetSpan.Value;
 
-            if (position < span.Start)
-            {
-                return span.Start - position;
-            }
-            else if (position > span.End)
-            {
-                return position - span.End;
-            }
-            else
-            {
-                return 0;
-            }
```

Comment:
i totally understood this logic.  if you're in the span, you're in it.  otherwise we look how far hte pos is from the start or end depending on if you're before it or after it.  i'm not sure i get the new calculation.

Reply:
I'll document it better. Essentially it's a sum of "how far from each other are the starts" and "how far from each other are the ends". In case of empty selection it's just the sum of distance between caret & end and caret & beginning. In case of non-empty selection it allows distinction between e.g. `$[|        |]$` and `$[|     |]   $`

Reply:
I'll try to document it better.